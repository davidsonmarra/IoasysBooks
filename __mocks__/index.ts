import BookDTO from '../src/@types/BookDTO';

interface IMockFetchDataBooks {
  data: BookDTO[];
  page: number;
  totalPages: number;
  totalItems: number;
}

export const mockBook: BookDTO = {
  id: '1',
  title: 'Test Title',
  imageUrl: 'https://www.test.com',
  authors: ['Test Author'],
  pageCount: 100,
  publisher: 'Test Publisher',
  published: 1980,
  category: 'Test Category',
  language: 'Test Language',
  isbn10: 'Test ISBN10',
  isbn13: 'Test ISBN13',
  description: 'Test Description'
};

export const mockPreloadedStateEmpty = {
  profile: {
    email: '',
    token: '',
    error: {} as Error,
    isLogged: false,
    isLoading: false
  },
  books: {
    isEnd: false,
    loadingFetchBooks: false,
    error: {} as Error,
    booksData: [],
    category: { key: '', title: '' },
    search: ''
  }
};

export const mockPreloadedStateFilled = {
  profile: {
    email: 'davidson',
    token: 'teste',
    error: {} as Error,
    isLogged: true,
    isLoading: false
  },
  books: {
    isEnd: false,
    loadingFetchBooks: false,
    error: {} as Error,
    booksData: [mockBook],
    category: { key: '', title: '' },
    search: ''
  }
};

export const mockFetchDataBooks: IMockFetchDataBooks = {
  data: [
    {
      ...mockBook,
      id: '2',
      title: 'Harry Potter 1'
    },
    {
      ...mockBook,
      id: '3',
      title: 'Harry Potter 2'
    },
    {
      ...mockBook,
      id: '4',
      title: 'Harry Potter 3'
    },
    {
      ...mockBook,
      id: '5',
      title: 'Harry Potter 4'
    },
    {
      ...mockBook,
      id: '6',
      title: 'Harry Potter 5'
    }
  ],
  page: 1,
  totalPages: 1,
  totalItems: 2
};
