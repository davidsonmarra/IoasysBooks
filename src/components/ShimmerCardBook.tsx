import React from 'react';
import styled from 'styled-components/native';
import { RFValue } from 'react-native-responsive-fontsize';
import Animated, { FadeIn } from 'react-native-reanimated';

export function ShimmerCardBook() {
  return (
    <AnimatedWrapper entering={FadeIn}>
      <StyledContainer>
        <StyledImageBook />
        <StyledInfo>
          <StyledHeader>
            <StyledTitle />
            <StyledAuthor />
          </StyledHeader>
          <StyledFooter>
            <StyledInfoDetails />
            <StyledInfoDetails />
            <StyledInfoDetails />
          </StyledFooter>
        </StyledInfo>
      </StyledContainer>
    </AnimatedWrapper>
  );
}

const StyledWrapper = styled.View`
  width: 100%;
  padding: 0 ${RFValue(16)}px;
`;

const AnimatedWrapper = Animated.createAnimatedComponent(StyledWrapper);

const StyledContainer = styled.View`
  width: 100%;
  height: ${RFValue(180)}px;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  margin-bottom: ${RFValue(16)}px;
  padding: ${RFValue(8)}px ${RFValue(16)}px;
  border-radius: ${RFValue(4)}px;
  box-shadow: 0px 6px 24px rgba(84, 16, 95, 0.13);
  background-color: ${({ theme }) => theme.colors.shape};
`;

const StyledImageBook = styled.Image`
  width: ${RFValue(81)}px;
  height: ${RFValue(122)}px;
  margin-right: ${RFValue(16)}px;
  border-radius: ${RFValue(2)}px;
  background-color: ${({ theme }) => theme.colors.shimmer};
`;

const StyledInfo = styled.View`
  flex: 1;
  height: ${RFValue(180)}px;
  justify-content: space-between;
  padding: ${RFValue(16)}px 0;
`;

const StyledHeader = styled.View``;

const StyledTitle = styled.View`
  width: 60%;
  height: ${RFValue(12)}px;
  border-radius: ${RFValue(2)}px;
  margin-bottom: ${RFValue(2)}px;
  background-color: ${({ theme }) => theme.colors.shimmer};
`;

const StyledAuthor = styled.View`
  width: 45%;
  height: ${RFValue(12)}px;
  border-radius: ${RFValue(2)}px;
  background-color: ${({ theme }) => theme.colors.shimmer};
`;

const StyledFooter = styled.View``;

const StyledInfoDetails = styled.View`
  width: 45%;
  height: ${RFValue(12)}px;
  border-radius: ${RFValue(2)}px;
  margin-bottom: ${RFValue(4)}px;
  background-color: ${({ theme }) => theme.colors.shimmer};
`;
