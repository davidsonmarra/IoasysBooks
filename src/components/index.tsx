export { Input } from './form/Input';
export { SearchInput } from './form/SearchInput';
export { FilterModal } from './FilterModal';
export { CardBook } from './CardBook';
export { ShimmerCardBook } from './ShimmerCardBook';
