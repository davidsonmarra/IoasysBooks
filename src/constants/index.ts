const constants = {
  BASE_URL: 'https://books.ioasys.com.br/api/v1',
  TIMEOUT: 8000,
  asyncStorageUserKey: '@books_asyncStorageUserKey',
  asyncStorageUserId: '@books_asyncStorageUserId',
  asyncStorageUserRefresh: '@books_asyncStorageUserRefresh'
};

export default constants;
