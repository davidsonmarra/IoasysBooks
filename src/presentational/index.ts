export { SignIn } from './SignIn';
export { BookDetails } from './BookDetails';
export { Home } from './Home';
