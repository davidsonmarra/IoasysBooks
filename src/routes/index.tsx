import React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { PublicRoutes } from './stacks/public.routes';
import { useSelector } from 'react-redux';
import { IRootState } from '../store/store';
import { AuthRoutes } from './stacks/auth.routes';

export function Routes() {
  const { isLogged } = useSelector(({ profile }: IRootState) => profile);

  return <NavigationContainer>{isLogged ? <AuthRoutes /> : <PublicRoutes />}</NavigationContainer>;
}
