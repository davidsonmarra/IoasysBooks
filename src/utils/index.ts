import { AxiosError } from 'axios';

export const isObjectEmpty = (prop: Record<string, unknown> | Error | AxiosError<unknown, any>) =>
  Object.keys(prop).length > 0;

export { categories } from './categories';
