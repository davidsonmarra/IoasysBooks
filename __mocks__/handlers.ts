import { rest } from 'msw';
import { mockFetchDataBooks, mockFetchDataRefresh, mockFetchDataSignIn } from '.';
import constants from '../src/constants';

export const handlers = [
  rest.get(`${constants.BASE_URL}/books`, (req, res, ctx) => {
    const search: string = req.url.searchParams.get('title') || '';
    return search.includes('error')
      ? res(
          ctx.status(401),
          ctx.json({
            errorMessage: `Error 401`
          })
        )
      : res(ctx.json(mockFetchDataBooks));
  }),
  rest.post(`${constants.BASE_URL}/auth/sign-in`, async (req, res, ctx) => {
    const { email } = await req.json();
    return email.includes('error')
      ? res(
          ctx.status(401),
          ctx.json({
            errorMessage: `Error 401`
          })
        )
      : res(ctx.json(mockFetchDataSignIn));
  }),
  rest.post(`${constants.BASE_URL}/auth/refresh-token`, async (req, res, ctx) => {
    const { refreshToken } = await req.json();
    return refreshToken.includes('error')
      ? res(
          ctx.status(401),
          ctx.json({
            errorMessage: `Error 401`
          })
        )
      : res(ctx.json(mockFetchDataRefresh));
  })
];
