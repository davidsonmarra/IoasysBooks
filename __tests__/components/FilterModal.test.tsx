import { FilterModal } from '../../src/components';
import { mockPreloadedStateFilled } from '../../__mocks__';
import {
  matchSnapshotWithProviders,
  fireEvent,
  renderWithCustomProviders,
  cleanup
} from '../test-utils';
import { server } from '../jest.setup';

const params = (visible = true, handleModal = jest.fn(), setOffset = jest.fn(), search = '') => ({
  visible,
  handleModal,
  setOffset,
  search
});

describe('Filter Modal', () => {
  afterEach(() => {
    cleanup();
    server.resetHandlers();
  });

  afterAll(() => server.close());

  it('should match a snapshot', () => {
    matchSnapshotWithProviders(<FilterModal {...params()} />);
  });

  it('should select category correctly', () => {
    const {
      render: { getByTestId, getByLabelText },
      store
    } = renderWithCustomProviders(<FilterModal {...params()} />, mockPreloadedStateFilled);
    const categoryButton = getByTestId(/category-tales-button/i);
    const buttonFilterSubmit = getByLabelText(/submit filter/i);

    fireEvent.press(categoryButton);
    fireEvent.press(buttonFilterSubmit);
    expect(store.getState().books.category).toEqual({ key: 'tales', title: 'Contos' });

    fireEvent.press(categoryButton);
    fireEvent.press(buttonFilterSubmit);
    expect(store.getState().books.category).toEqual({});
  });
});
