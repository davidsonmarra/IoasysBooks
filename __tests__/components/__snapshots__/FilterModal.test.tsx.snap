// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter Modal should match a snapshot 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  onRequestClose={[MockFunction]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#00000052",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
    ]
  }
  testID="filter-modal"
  transparent={true}
  visible={true}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#00000052",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
          "paddingBottom": 0,
          "paddingLeft": 31,
          "paddingRight": 31,
          "paddingTop": 0,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#FFFFFF",
            "paddingBottom": 47,
            "paddingLeft": 31,
            "paddingRight": 31,
            "paddingTop": 31,
            "width": "100%",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-end",
            },
          ]
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "borderColor": "#00000052",
              "borderRadius": 31,
              "borderStyle": "solid",
              "borderWidth": 2,
              "height": 63,
              "justifyContent": "center",
              "opacity": 1,
              "width": 63,
            }
          }
        >
          <SvgMock
            height={31}
            style={
              Array [
                Object {},
              ]
            }
            width={31}
          />
        </View>
      </View>
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#333333",
                "fontFamily": "Heebo-Medium",
                "fontSize": 24,
                "marginBottom": 31,
              },
            ]
          }
        >
          Selecione a categoria
        </Text>
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
                "flexWrap": "wrap",
              },
            ]
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "borderColor": "#00000052",
                "borderRadius": 86,
                "borderStyle": "solid",
                "borderWidth": 2,
                "marginBottom": 16,
                "marginRight": 16,
                "opacity": 1,
                "paddingBottom": 12,
                "paddingLeft": 31,
                "paddingRight": 31,
                "paddingTop": 12,
              }
            }
            testID="category-biographies-button"
          >
            <Text
              isSelected={false}
              style={
                Array [
                  Object {
                    "color": "#333333",
                    "fontFamily": "Heebo-Regular",
                    "fontSize": 24,
                  },
                ]
              }
            >
              Biografia
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "borderColor": "#00000052",
                "borderRadius": 86,
                "borderStyle": "solid",
                "borderWidth": 2,
                "marginBottom": 16,
                "marginRight": 16,
                "opacity": 1,
                "paddingBottom": 12,
                "paddingLeft": 31,
                "paddingRight": 31,
                "paddingTop": 12,
              }
            }
            testID="category-collections-button"
          >
            <Text
              isSelected={false}
              style={
                Array [
                  Object {
                    "color": "#333333",
                    "fontFamily": "Heebo-Regular",
                    "fontSize": 24,
                  },
                ]
              }
            >
              Coleções
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "borderColor": "#00000052",
                "borderRadius": 86,
                "borderStyle": "solid",
                "borderWidth": 2,
                "marginBottom": 16,
                "marginRight": 16,
                "opacity": 1,
                "paddingBottom": 12,
                "paddingLeft": 31,
                "paddingRight": 31,
                "paddingTop": 12,
              }
            }
            testID="category-behavior-button"
          >
            <Text
              isSelected={false}
              style={
                Array [
                  Object {
                    "color": "#333333",
                    "fontFamily": "Heebo-Regular",
                    "fontSize": 24,
                  },
                ]
              }
            >
              Comportamento
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "borderColor": "#00000052",
                "borderRadius": 86,
                "borderStyle": "solid",
                "borderWidth": 2,
                "marginBottom": 16,
                "marginRight": 16,
                "opacity": 1,
                "paddingBottom": 12,
                "paddingLeft": 31,
                "paddingRight": 31,
                "paddingTop": 12,
              }
            }
            testID="category-tales-button"
          >
            <Text
              isSelected={false}
              style={
                Array [
                  Object {
                    "color": "#333333",
                    "fontFamily": "Heebo-Regular",
                    "fontSize": 24,
                  },
                ]
              }
            >
              Contos
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "borderColor": "#00000052",
                "borderRadius": 86,
                "borderStyle": "solid",
                "borderWidth": 2,
                "marginBottom": 16,
                "marginRight": 16,
                "opacity": 1,
                "paddingBottom": 12,
                "paddingLeft": 31,
                "paddingRight": 31,
                "paddingTop": 12,
              }
            }
            testID="category-literary-criticism-button"
          >
            <Text
              isSelected={false}
              style={
                Array [
                  Object {
                    "color": "#333333",
                    "fontFamily": "Heebo-Regular",
                    "fontSize": 24,
                  },
                ]
              }
            >
              Crítica Literária
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "borderColor": "#00000052",
                "borderRadius": 86,
                "borderStyle": "solid",
                "borderWidth": 2,
                "marginBottom": 16,
                "marginRight": 16,
                "opacity": 1,
                "paddingBottom": 12,
                "paddingLeft": 31,
                "paddingRight": 31,
                "paddingTop": 12,
              }
            }
            testID="category-scienceFiction-button"
          >
            <Text
              isSelected={false}
              style={
                Array [
                  Object {
                    "color": "#333333",
                    "fontFamily": "Heebo-Regular",
                    "fontSize": 24,
                  },
                ]
              }
            >
              Ficção Científica
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "borderColor": "#00000052",
                "borderRadius": 86,
                "borderStyle": "solid",
                "borderWidth": 2,
                "marginBottom": 16,
                "marginRight": 16,
                "opacity": 1,
                "paddingBottom": 12,
                "paddingLeft": 31,
                "paddingRight": 31,
                "paddingTop": 12,
              }
            }
            testID="category-folklore-button"
          >
            <Text
              isSelected={false}
              style={
                Array [
                  Object {
                    "color": "#333333",
                    "fontFamily": "Heebo-Regular",
                    "fontSize": 24,
                  },
                ]
              }
            >
              Folclore
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "borderColor": "#00000052",
                "borderRadius": 86,
                "borderStyle": "solid",
                "borderWidth": 2,
                "marginBottom": 16,
                "marginRight": 16,
                "opacity": 1,
                "paddingBottom": 12,
                "paddingLeft": 31,
                "paddingRight": 31,
                "paddingTop": 12,
              }
            }
            testID="category-genealogy-button"
          >
            <Text
              isSelected={false}
              style={
                Array [
                  Object {
                    "color": "#333333",
                    "fontFamily": "Heebo-Regular",
                    "fontSize": 24,
                  },
                ]
              }
            >
              Genealogia
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "borderColor": "#00000052",
                "borderRadius": 86,
                "borderStyle": "solid",
                "borderWidth": 2,
                "marginBottom": 16,
                "marginRight": 16,
                "opacity": 1,
                "paddingBottom": 12,
                "paddingLeft": 31,
                "paddingRight": 31,
                "paddingTop": 12,
              }
            }
            testID="category-humor-button"
          >
            <Text
              isSelected={false}
              style={
                Array [
                  Object {
                    "color": "#333333",
                    "fontFamily": "Heebo-Regular",
                    "fontSize": 24,
                  },
                ]
              }
            >
              Humor
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "borderColor": "#00000052",
                "borderRadius": 86,
                "borderStyle": "solid",
                "borderWidth": 2,
                "marginBottom": 16,
                "marginRight": 16,
                "opacity": 1,
                "paddingBottom": 12,
                "paddingLeft": 31,
                "paddingRight": 31,
                "paddingTop": 12,
              }
            }
            testID="category-children-button"
          >
            <Text
              isSelected={false}
              style={
                Array [
                  Object {
                    "color": "#333333",
                    "fontFamily": "Heebo-Regular",
                    "fontSize": 24,
                  },
                ]
              }
            >
              Infantojuvenis
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "borderColor": "#00000052",
                "borderRadius": 86,
                "borderStyle": "solid",
                "borderWidth": 2,
                "marginBottom": 16,
                "marginRight": 16,
                "opacity": 1,
                "paddingBottom": 12,
                "paddingLeft": 31,
                "paddingRight": 31,
                "paddingTop": 12,
              }
            }
            testID="category-games-button"
          >
            <Text
              isSelected={false}
              style={
                Array [
                  Object {
                    "color": "#333333",
                    "fontFamily": "Heebo-Regular",
                    "fontSize": 24,
                  },
                ]
              }
            >
              Jogos
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "borderColor": "#00000052",
                "borderRadius": 86,
                "borderStyle": "solid",
                "borderWidth": 2,
                "marginBottom": 16,
                "marginRight": 16,
                "opacity": 1,
                "paddingBottom": 12,
                "paddingLeft": 31,
                "paddingRight": 31,
                "paddingTop": 12,
              }
            }
            testID="category-newspapers-button"
          >
            <Text
              isSelected={false}
              style={
                Array [
                  Object {
                    "color": "#333333",
                    "fontFamily": "Heebo-Regular",
                    "fontSize": 24,
                  },
                ]
              }
            >
              Jornais
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "borderColor": "#00000052",
                "borderRadius": 86,
                "borderStyle": "solid",
                "borderWidth": 2,
                "marginBottom": 16,
                "marginRight": 16,
                "opacity": 1,
                "paddingBottom": 12,
                "paddingLeft": 31,
                "paddingRight": 31,
                "paddingTop": 12,
              }
            }
            testID="category-brazilian-literature-button"
          >
            <Text
              isSelected={false}
              style={
                Array [
                  Object {
                    "color": "#333333",
                    "fontFamily": "Heebo-Regular",
                    "fontSize": 24,
                  },
                ]
              }
            >
              Literatura Brasileira
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "borderColor": "#00000052",
                "borderRadius": 86,
                "borderStyle": "solid",
                "borderWidth": 2,
                "marginBottom": 16,
                "marginRight": 16,
                "opacity": 1,
                "paddingBottom": 12,
                "paddingLeft": 31,
                "paddingRight": 31,
                "paddingTop": 12,
              }
            }
            testID="category-foreign-literature-button"
          >
            <Text
              isSelected={false}
              style={
                Array [
                  Object {
                    "color": "#333333",
                    "fontFamily": "Heebo-Regular",
                    "fontSize": 24,
                  },
                ]
              }
            >
              Literatura Estrangeira
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "borderColor": "#00000052",
                "borderRadius": 86,
                "borderStyle": "solid",
                "borderWidth": 2,
                "marginBottom": 16,
                "marginRight": 16,
                "opacity": 1,
                "paddingBottom": 12,
                "paddingLeft": 31,
                "paddingRight": 31,
                "paddingTop": 12,
              }
            }
            testID="category-rare-books-button"
          >
            <Text
              isSelected={false}
              style={
                Array [
                  Object {
                    "color": "#333333",
                    "fontFamily": "Heebo-Regular",
                    "fontSize": 24,
                  },
                ]
              }
            >
              Livros Raros
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "borderColor": "#00000052",
                "borderRadius": 86,
                "borderStyle": "solid",
                "borderWidth": 2,
                "marginBottom": 16,
                "marginRight": 16,
                "opacity": 1,
                "paddingBottom": 12,
                "paddingLeft": 31,
                "paddingRight": 31,
                "paddingTop": 12,
              }
            }
            testID="category-manuscripts-button"
          >
            <Text
              isSelected={false}
              style={
                Array [
                  Object {
                    "color": "#333333",
                    "fontFamily": "Heebo-Regular",
                    "fontSize": 24,
                  },
                ]
              }
            >
              Manuscritos
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "borderColor": "#00000052",
                "borderRadius": 86,
                "borderStyle": "solid",
                "borderWidth": 2,
                "marginBottom": 16,
                "marginRight": 16,
                "opacity": 1,
                "paddingBottom": 12,
                "paddingLeft": 31,
                "paddingRight": 31,
                "paddingTop": 12,
              }
            }
            testID="category-poetry-button"
          >
            <Text
              isSelected={false}
              style={
                Array [
                  Object {
                    "color": "#333333",
                    "fontFamily": "Heebo-Regular",
                    "fontSize": 24,
                  },
                ]
              }
            >
              Poesia
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "borderColor": "#00000052",
                "borderRadius": 86,
                "borderStyle": "solid",
                "borderWidth": 2,
                "marginBottom": 16,
                "marginRight": 16,
                "opacity": 1,
                "paddingBottom": 12,
                "paddingLeft": 31,
                "paddingRight": 31,
                "paddingTop": 12,
              }
            }
            testID="category-another-subjects-button"
          >
            <Text
              isSelected={false}
              style={
                Array [
                  Object {
                    "color": "#333333",
                    "fontFamily": "Heebo-Regular",
                    "fontSize": 24,
                  },
                ]
              }
            >
              Outros
            </Text>
          </View>
        </View>
      </View>
      <View
        accessibilityLabel="submit filter"
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignSelf": "center",
            "borderColor": "#B22E6F",
            "borderRadius": 86,
            "borderStyle": "solid",
            "borderWidth": 2,
            "marginTop": 94,
            "opacity": 1,
            "paddingBottom": 24,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 24,
            "width": 177,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "alignSelf": "center",
                "color": "#B22E6F",
                "fontFamily": "Heebo-Medium",
              },
            ]
          }
        >
          Filtrar
        </Text>
      </View>
    </View>
  </View>
</Modal>
`;
